Slim/Control/Commands.pm

```diff
@@ -1987,7 +1988,7 @@ sub playlistcontrolCommand {
 			return;
 		}
 
-	} elsif (defined(my $work_id = $request->getParam('work_id'))) {
+	} elsif (defined(my $work_id = $request->getParam('work_id')) && $request->getParam('work_id') ne '-1') {
```

Comment:
Read `work_id` once at the top and simplify all these checks for it. Set it to an empty string if undefined to have really simple checks here.